src/Rule/BrokenLink.php

```diff
@@ -19,47 +19,38 @@ public function id()
 
 	public function check()
 	{
-		$links = array();
 		foreach ($this->getAllElements('a') as $a) {
 			$href = $a->getAttribute('href');
 			if ($href) {
-				$links[$href] = $a;
+				$this->checkLink($a, $href);
 			}
             $this->totalTests++;
 		}
-		$this->checkLink($links);
 
 		return count($this->issues);
 	}
 
-	private function checkLink($links) {
-		$curls = array();
-		$mcurl = curl_multi_init();
-		foreach (array_keys($links) as $i => $link) {
-			$curls[$i] = curl_init();
-			curl_setopt($curls[$i], CURLOPT_URL, $link);
-			curl_setopt($curls[$i], CURLOPT_HEADER, true);
-			curl_setopt($curls[$i], CURLOPT_NOBODY, true);
-			curl_setopt($curls[$i], CURLOPT_REFERER, true);
-			curl_setopt($curls[$i], CURLOPT_TIMEOUT, 2);
-			curl_setopt($curls[$i], CURLOPT_TIMEOUT, 2);
-			curl_setopt($curls[$i], CURLOPT_AUTOREFERER, true);
-			curl_setopt($curls[$i], CURLOPT_RETURNTRANSFER, true);
-			curl_setopt($curls[$i], CURLOPT_FOLLOWLOCATION, true);
-			curl_multi_add_handle($mcurl, $curls[$i]);
-		}
+	private function checkLink($element, $link) {
+		$curl = curl_init();
+		
+		curl_setopt($curl, CURLOPT_URL, $link);
+		curl_setopt($curl, CURLOPT_HEADER, true);
+		curl_setopt($curl, CURLOPT_NOBODY, true);
+		curl_setopt($curl, CURLOPT_REFERER, true);
+		curl_setopt($curl, CURLOPT_TIMEOUT, 2);
+		curl_setopt($curl, CURLOPT_AUTOREFERER, true);
+		curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
+		
 		$running = null;
```

Comment:
Let's remove the $running var and the do/while loop. I don't think those are doing anything.